src/components/query/visual-query-builder/InterpolatedResolutionSettings.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { SelectableValue } from '@grafana/data';
+import { SiteWiseResolution, AssetPropertyInterpolatedQuery } from 'types';
+import { Select } from '@grafana/ui';
+import { EditorField, EditorFieldGroup } from '@grafana/plugin-ui';
+import { useOptionsWithVariables } from 'common/useOptionsWithVariables';
+
+const RESOLUTIONS: Array<SelectableValue<SiteWiseResolution>> = [
+  {
+    value: SiteWiseResolution.Auto,
+    label: 'Auto',
+    description:
+      'Picks a resolution based on the time window. ' +
+      'Will switch to raw data if higher than 1m resolution is needed',
+  },
+  { value: SiteWiseResolution.Sec, label: 'Second', description: '1 point every second' },
+  { value: SiteWiseResolution.TenSec, label: '10 Seconds', description: '1 point every 10 seconds' },
+  { value: SiteWiseResolution.Min, label: 'Minute', description: '1 point every minute' },
+  { value: SiteWiseResolution.TenMin, label: '10 Minutes', description: '1 point every 10 minutes' },
+  { value: SiteWiseResolution.Hour, label: 'Hour', description: '1 point every hour' },
+  { value: SiteWiseResolution.TenHour, label: '10 Hours', description: '1 point every 10 hours' },
+  { value: SiteWiseResolution.Day, label: 'Day', description: '1 point every day' },
+];
+
+export const InterpolatedResolutionSettings = ({
+  onChange,
+  query,
+}: {
+  query: AssetPropertyInterpolatedQuery;
+  onChange: (value: AssetPropertyInterpolatedQuery) => void;
+}) => {
+  const onResolutionChange = (sel: SelectableValue<string>) => {
+    onChange({ ...query, resolution: sel.value as SiteWiseResolution });
```

Comment:
We're casting to SitewiseResolution here, but it's possible that the variable value is a string that isn't a valid resolution option. Do you think it would be worth it to add some kind of validation here (like an inline error) in that case (and otherwise we'd do a type guard)?

Reply:
the challenge is it can be any sitewise resolution or a template variable, and there's no way i'm aware of to differentiate between a valid template variable expression vs just a random string

Reply:
aren't these the only valid strings for resolution: https://github.com/grafana/iot-sitewise-datasource/blob/5231a1c691a30cbfbac1ba7acb59e6888d2eae58/src/types.ts#L42?

Reply:
yes, after the template gets resolved. With my change, `$resolution` for example would be a valid string input into the component

Reply:
We could resolve it here, but if you think it's not worth the work, I'm happy to keep this